enhancements/microshift/rhoai-model-serving-on-microshift.md

```diff
@@ -0,0 +1,373 @@
+
+## Proposal
+
+Extract kserve manifests from RHOAI Operator image and adjust them for MicroShift:
+- Make sure that cert-manager is not required, instead leverage OpenShift's service-ca.
+  - This might require adding some extra annotations to resources so the service-ca injects the certs.
+- Drop requirement for Istio as an Ingress Controller and use OpenShift Router instead.
+  - Done by changing kserve's setting configmap to use another ingress controller.
+  - RHOAI disables automatic Ingress creation by kserve and odh-model-controller takes over creation of appropriate CRs.
+    We need to decide if we want to re-enable it or instruct user to create Route CR themselves.
+- Use 'RawDeployment' mode, so that neither Service Mesh nor Serverless are required,
+  to minimize to make the solution suitable for edge devices.
+  - Also done in the configmap.
+- Package the manifests as an RPM `microshift-kserve` for usage.
+
+Provide users with ServingRuntime definitions derived from RHOAI, so they
+are not forced to use upstream manifests.
+Decision on how to do this is pending. See open questions.
+
+### Rebase procedure in detail
+
+1. `rebase_rhoai.sh` script shall take an argument pointing to the RHOAI Operator Bundle Image.
+1. The script will:
+   1. Extract ClusterServiceVersion (CSV) of the RHOAI Operator.
+   1. Obtain RHOAI Operator image reference from the CSV.
+   1. Extract /opt/manifests from the RHOAI Operator image.
+   1. Copy contents of /opt/manifests/kserve to MicroShift repository.
+   1. Copy contents of /opt/manifests/odh-model-controller/runtimes to MicroShift repository.
+   1. Prepare top-level kustomization.yaml with MicroShift-specific tweaks.
+      1. Create kserve settings ConfigMap to configure MicroShift-specific tweaks.
+      1. Use contents of different files from RHOAI Operator's /opt/manifests
+         to get downstream-rebuilt image references and apply them over the
+         existing manifests.
+
+
+### Workflow Description
+
+**User** is a human administrating and using MicroShift cluster/device.
+
+(RPM vs ostree vs bootc is skipped because it doesn't differ from any other MicroShift's RPM).
+
+1. User installs `microshift-kserve` RPM and restarts MicroShift service.
+1. Kserve manifest are deployed.
+1. User configures the hardware, the OS, and additional Kubernetes components to
+   make use of their accelerators.
+1. ServingRuntimes are delivered with the kserve RPM or deployed by the user.
+1. User creates InferenceService CR which references ServingRuntime of their choice
+   and reference to the model.
+1. Kserve creates Deployment, Ingress, and other.
+1. Resources from previous step become ready and user can make HTTP/GRPC calls
+   to the model server.
+
+### API Extensions
+
+`microshift-kserve` RPM will bring following CRDs, however they're not becoming
+part of the core MicroShift deployment:
```

Comment:
@pmtk, is [ClusterServingRuntimes](https://github.com/red-hat-data-services/kserve/blob/master/config/crd/full/serving.kserve.io_clusterservingruntimes.yaml) missing from this list?

Reply:
Not decided yet, see `### How to deliver ServingRuntime CRs`